src/DotLiquid/StandardFilters.cs

```diff
@@ -455,13 +451,14 @@ private static IEnumerable SortInternal(StringComparer stringComparer, object in
             {
                 ary.Sort((a, b) => comparer.Compare(a, b));
             }
-            else if ((ary.All(o => o is IDictionary)) && (ary.Any(o => ((IDictionary)o).Contains(property))))
-            {
-                ary.Sort((a, b) => comparer.Compare(((IDictionary)a)[property], ((IDictionary)b)[property]));
-            }
-            else if (ary.All(o => o.RespondTo(property)))
+            else
             {
-                ary.Sort((a, b) => comparer.Compare(a.Send(property), b.Send(property)));
+                ary.Sort((a, b) =>
+                {
+                    var aPropertyValue = ResolveObjectPropertyValue(a, property);

```

Comment:
For readability reason, when using static method, I suggest putting the class name prefix, e.g. StandardFilters.ResolveObjectPropertyValue(a, property);